src/jdk.incubator.code/share/classes/jdk/incubator/code/interpreter/Interpreter.java

```diff
@@ -227,15 +227,15 @@ Block exception(MethodHandles.Lookup l, Throwable e) {
         }
     }
 
-    static final class VarBox
+    public static final class VarBox
```

Comment:
You don't need to make this public. Look what it implements and provide your own implementation.

Separately, we likely need to address how we expose a runtime value correspond to a symbolic `Var` value. We will encounter similar issues with tuples and funcs, so we need some general mapping i expect. This may become easier to identify as we refactor the code base more cleanly into dialects.